pages/[network]/avatar/[address].tsx

```diff
@@ -119,21 +124,22 @@ export const getServerSideProps: GetServerSideProps<AvatarPageProps> = async (
 
     const nodeType = getNodeType(network);
     const networkType = getNetworkType(network);
-    const avatarJsonObj = await getAvatar(nodeType, networkType, address);
-
     const sdk = getSdk(networkType, nodeType);
-    const inventoryJsonObj = await sdk.inventory(address);
+    const avatarJsonObj = await sdk.avatar(address);
+    const inventoryJsonObj = avatarJsonObj.inventory;
     const inventoryObj = parseToInventory(inventoryJsonObj);
-    const balanceJsonObjs = await Promise.all(
-        CURRENCY_TICKERS.map(
-            (currencyTicker, index) => getBalance(
-                nodeType,
-                networkType,
-                index <= 0 ? avatarJsonObj.agentAddress : address,
-                currencyTicker
+    const balanceJsonObjs = avatarJsonObj.agentAddress === null
+        ? []
+        : await Promise.all(
+            CURRENCY_TICKERS.map(
+                (currencyTicker, index) => getBalance(
```

Comment:
If the below review comment is accepted, the `index` argument is unnecessary:

```suggestion
                (currencyTicker) => getBalance(
```